### ISSUE TITLE
No error diff color on forks pool

### ISSUE THREAD

**Issue by hi-ogawa** [vitest-dev/vitest#7042]
### Describe the bug

Not sure why I just noticed this now, but it looks like we don't have diff color on forks.

https://stackblitz.com/edit/vitest-dev-vitest-cnfjynoc?file=test%2Fbasic.test.ts

`npm run test`

![image](https://github.com/user-attachments/assets/30c99cd8-f0d7-4aa9-b535-72a367b5a8e3)

`npm run test  --pool threads`

![image](https://github.com/user-attachments/assets/3301bfdd-d798-4b70-af5d-4f99aa389fd1)

### Reproduction

https://stackblitz.com/edit/vitest-dev-vitest-cnfjynoc?file=test%2Fbasic.test.ts

### System Info

```shell
System:
    OS: Linux 5.0 undefined
    CPU: (8) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 0 Bytes / 0 Bytes
    Shell: 1.0 - /bin/jsh
  Binaries:
    Node: 18.20.3 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 10.2.3 - /usr/local/bin/npm
    pnpm: 8.15.6 - /usr/local/bin/pnpm
  npmPackages:
    vitest: 2.1.8 => 2.1.8
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by hi-ogawa** [comment]
It turned out this is due to tinypool v1.0.2 https://github.com/tinylibs/tinypool/pull/104.

- forks: tinypool v1.0.1 https://stackblitz.com/edit/vitest-dev-vitest-hhruvape?file=test%2Fsuite.test.js

![image](https://github.com/user-attachments/assets/33c4c38c-10fb-4ec2-af29-24ffc95a1b0c)

- forks: tinypool v1.0.2 https://stackblitz.com/edit/vitest-dev-vitest-gxjfg4h7?file=package.json

![image](https://github.com/user-attachments/assets/d8c9809b-9790-41de-a894-d19c1637f4c4)

- threads: : tinypool v1.0.2

![image](https://github.com/user-attachments/assets/d15f15bc-6289-4d94-8cd6-39d25f0e03c0)

@AriPerkkio Do you have some idea for fix? `tty` becoming `false` for users is not probably crucial per-se, but our own color logic not working on default setup seems bad.

**Comment by AriPerkkio** [comment]
Good catch @hi-ogawa, I wasn't expecting this.

Here's minimal reproduction without Tinypool demonstrating the fix from https://github.com/tinylibs/tinypool/pull/104. Happy to change the implementation another approach is found:

<details open>
  <summary>Intercept stdout of node:child_process in main process</summary>

```js
import { writeFileSync } from "node:fs";
import { fork } from "node:child_process";
import { setTimeout } from "node:timers/promises";

writeFileSync(
  "./worker.mjs",
  `
process.stdout.write("Message\\n");
process.stderr.write("Error message\\n");
  `.trim(),
  "utf8"
);

const write = process.stdout.write.bind(process.stdout);

process.stdout.write = (data) => {
  write(`Captured stdout: ${data}`);
};

process.stderr.write = (data) => {
  write(`Captured stderr: ${data}`);
};

{
  // Unable to intercept stdout
  const subprocess = fork("./worker.mjs");

  await setTimeout(3_000);
  subprocess.kill();
}

{
  // Intercepts stdout
  const subprocess = fork("./worker.mjs", { stdio: "pipe" });
  subprocess.stdout.pipe(process.stdout);
  subprocess.stderr.pipe(process.stderr);

  await setTimeout(3_000);
  subprocess.kill();
}
```

</details>

<details>
  <summary>Interceps stdout of node:worker_threads, works automatically</summary>

```js
import { writeFileSync } from "node:fs";
import { Worker } from "node:worker_threads";
import { setTimeout } from "node:timers/promises";

writeFileSync(
  "./worker.mjs",
  `
process.stdout.write("Message\\n");
process.stderr.write("Error message\\n");
  `.trim(),
  "utf8"
);

const write = process.stdout.write.bind(process.stdout);

process.stdout.write = (data) => {
  write(`Captured stdout: ${data}`);
};

process.stderr.write = (data) => {
  write(`Captured stderr: ${data}`);
};

const worker = new Worker("./worker.mjs");

await setTimeout(3_000);
await worker.terminate();
```

</details>

**Comment by hi-ogawa** [comment]
I guess we could pass down some flag from main thread to test runtime and create `tinyrainbow` instance manually by `createColors(process.env.__VITEST_ISATTY__ === 'true')` to at least our own color.

**Comment by AriPerkkio** [comment]
Yup that should work! It makes sense that TTY checks start to fail after changing the streams. Not sure why colors should be disabled when TTY is disabled though.

**Comment by AriPerkkio** [comment]
@sheremet-va what do you think about having `process.env.FORCE_TTY` here? https://github.com/tinylibs/tinyrainbow/blob/c8202ffc00d9f7f039ebdc2af52d08c6376fb6c0/src/index.ts#L83-L97

It could be used in cases like this, where main thread/process is TTY but spawned processes' streams aren't.
